### PR TITLE
Add search functionality onto the all required tabs

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/Search/Search.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <div class="search mdc-text-field mdc-text-field--outlined mdc-text-field--with-leading-icon">
-  <i class="search__icon material-icons mdc-text-field__icon">search</i>
-  <input type="text" placeholder="Search..." class="mdc-text-field__input">
+  <!--<i class="search__icon material-icons mdc-text-field__icon">search</i>-->
```
Comment (ihor4568) [inline src/components/Search/Search.html]: why do you need this comment?

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -2,8 +2,17 @@ import { MDCTextField } from "@material/textfield";
...
     this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  querySelectors() {
+    this.searchIcon = this.mountPoint.querySelector(".search__icon");
+  }
+
+  addEventListeners() {
+    this.searchIcon.addEventListener("click", this.props.onSearchQuery);
```
Comment (ihor4568) [inline src/components/Search/Search.js]: please use 'onInput' event and listen it on input itself - no need for button

File: src/components/Main/Helpers.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export function findAllOccurrences(array, string) {
```
Comment (ihor4568) [inline src/components/Main/Helpers.js]: just move this to services/Helpers. And make a static method

File: src/components/Main/Main.js [hunk 4]
```diff
@@ -47,6 +48,22 @@ export class MainComponent {
     window.addEventListener("popstate", this.handleStatePath.bind(this));
   }
 
+  handleSearchQuery(e) {
+    const searchQuery = e.target.nextElementSibling.value;
```
Comment (ihor4568) [inline src/components/Main/Main.js]: is it possible to move most of this logic either to search or my songs component?

File: src/components/Search/Search.html (hunk 1 above)
Comment (ghost) [reply src/components/Search/Search.html]: Temp.

File: src/components/Search/Search.js (hunk 2 above)
Comment (ghost) [reply src/components/Search/Search.js]: Temp. I think that it should be a form, so just for now to test functionality a added a button.

File: src/components/Main/Helpers.js (hunk 3 above)
Comment (ghost) [reply src/components/Main/Helpers.js]: Thx

**Pull request closed without merging**